tap_hubspot/client.py

```diff
@@ -155,6 +129,14 @@ def request(self,
             metrics_status_code = response.status_code
 
             timer.tags[metrics.Tag.http_status_code] = metrics_status_code
+        
+        if response.status_code == 403:
+            LOGGER.warn('Invalid Auth Exception - 403')
+            raise InvalidAuthException(response.text)
+            
+        if response.status_code == 429:
+            LOGGER.warn('Rate limit hit - 429')
+            raise Server429Error(response.text)
```

Comment:
The tap will end here and not continue. Should we also add this to the `backoff.on_exception` decorator?

Reply:
Makes sense! Can you give me a reference? I will add the decorator

Reply:
I used the documentation from https://github.com/tomasbasham/ratelimit as reference, to add the @sleep_and_retry decorator to avoid hitting the RateLimit Exception